src/AutoRest.CSharp/Common/Input/InputTypes/Examples/ExampleMockValueBuilder.cs

```diff
@@ -63,11 +62,16 @@ private static InputParameterExample BuildParameterExample(InputParameter parame
                     // when it is constant, it could have DefaultValue
                     value = InputExampleValue.Value(parameter.Type, parameter.DefaultValue.Value);
                 }
-                else if (parameter.Type is InputUnionType unionType && unionType.UnionItemTypes.First() is InputLiteralType literalType)
+                else if (parameter.Type is InputUnionType unionType && unionType.UnionItemTypes[0] is InputLiteralType literalType)
                 {
                     // or it could be a union of literal types
                     value = InputExampleValue.Value(parameter.Type, literalType.Value);
                 }
+                else if (parameter.Type is InputEnumType enumType && enumType.AllowedValues[0].Value is { } enumValue)
```

Comment:
what's the scenario does this handle?

Reply:
This is fixing this case: https://github.com/Azure/autorest.csharp/pull/4682/files#r1593497407
And the purpose here is intuitive - a constant with a type of enum, we take its first value as example value.
If we do not handle this, the else branch returns a null value, and in generated examples, null or default will be used as the parameter value.